docs/docs/components/expression-metadata.js

```diff
@@ -0,0 +1,754 @@
+import {
+    toString as typeToString,
+    NumberType,
+    StringType,
+    BooleanType,
+    ColorType,
+    ObjectType,
+    ValueType,
+    ErrorType,
+    CollatorType,
+    array
+} from '../../../tsc/src/expression/types';
+import CompoundExpression from '../../../tsc/src/expression/compound_expression';
+import {
+    Color,
+    validateRGBA,
+    typeOf,
+    toString as valueToString
+} from '../../../tsc/src/expression/values';
+
+// registers compound expressions
+import expressionRegister from '../../../tsc/src/expression/definitions/index';
+import RuntimeError from '../../../tsc/src/expression/runtime_error';
+
+function has(key, obj) {
+    return key in obj;
+}
+
+function get(key, obj) {
+    const v = obj[key];
+    return typeof v === 'undefined' ? null : v;
+}
+function rgba(ctx, [r, g, b, a]) {
+    r = r.evaluate(ctx);
+    g = g.evaluate(ctx);
+    b = b.evaluate(ctx);
+    const alpha = a ? a.evaluate(ctx) : 1;
+    const error = validateRGBA(r, g, b, alpha);
+    if (error) throw new RuntimeError(error);
+    return new Color(
+        (r / 255) * alpha,
+        (g / 255) * alpha,
+        (b / 255) * alpha,
+        alpha
+    );
+}
+
+function binarySearch(v, a, i, j) {
+    while (i <= j) {
+        const m = (i + j) >> 1;
+        if (a[m] === v) return true;
+        if (a[m] > v) j = m - 1;
+        else i = m + 1;
+    }
+    return false;
+}
+
+function varargs(type) {
+    return { type };
+}
+
+CompoundExpression.register(expressionRegister, {
+    error: [
+        ErrorType,
+        [StringType],
+        (ctx, [v]) => {
+            throw new RuntimeError(v.evaluate(ctx));
+        }
+    ],
+    typeof: [
+        StringType,
+        [ValueType],
+        (ctx, [v]) => typeToString(typeOf(v.evaluate(ctx)))
+    ],
+    'to-rgba': [
+        array(NumberType, 4),
+        [ColorType],
+        (ctx, [v]) => {
+            return v.evaluate(ctx).toArray();
+        }
+    ],
+    rgb: [ColorType, [NumberType, NumberType, NumberType], rgba],
```

Comment:
This part (all the expressions here) should be generated, otherwise there's a need to change this file every time we change the expressions.